py-polars/src/lazyframe/visitor/nodes.rs

```
@@ -291,6 +291,14 @@ pub(crate) fn into_py(py: Python<'_>, plan: &IR) -> PyResult<PyObject> {
             predicate: predicate.into(),
         }
         .into_py(py),
+        IR::Scan {
+            hive_parts: Some(_),
+            ..
+        } => {
+            return Err(PyNotImplementedError::new_err(
+                "scan with hive partitioning",
+            ))
```

previously we were silently discarding this information.